File: 프로그래머스/2/140107. 점 찍기/점 찍기.py
import math

def solution(k, d):
    answer = 0
    
    def maxY(x):
        return math.floor(math.sqrt(d * d - x * x))
    
    for x in range(0, d + 1, k):
        answer += (math.floor(maxY(x) / k) + 1)
    
    return answer

File: 프로그래머스/2/140107. 점 찍기/점 찍기.js
function solution(k, d) {
  let answer = 0;
    
  for (let x = 0; x <= d; x += k) {
      answer += Math.ceil(Math.sqrt(d*d - x*x)/k) + (Math.sqrt(d*d - x*x) % k === 0 ? 1 : 0)
  }
    
  return answer;
}